docs/guide/acquiring_data/helioviewer.rst

```diff
@@ -30,52 +30,100 @@ Let's begin by getting a list of data sources available on the server
 using the get_datasources method::
 
     >>> from sunpy.net.helioviewer import HelioviewerClient
-
     >>> hv = HelioviewerClient()
     >>> datasources = hv.get_data_sources()  # doctest: +REMOTE_DATA
-
-    >>> # print a list of datasources and their associated ids
-    >>> for observatory, instruments in datasources.items():  # doctest: +REMOTE_DATA
-    ...     for inst, detectors in instruments.items():  # doctest: +REMOTE_DATA
-    ...         for det, measurements in detectors.items():  # doctest: +REMOTE_DATA
-    ...             for meas, params in measurements.items():  # doctest: +REMOTE_DATA
-    ...                 print("%s %s: %d" % (observatory, params['nickname'], params['sourceId']))  # doctest: +REMOTE_DATA
-    SOHO EIT 171: 0
-    SOHO EIT 195: 1
-    SOHO EIT 284: 2
-    SOHO EIT 304: 3
-    SOHO LASCO C2: 4
-    SOHO LASCO C3: 5
-    SOHO MDI Mag: 6
-    SOHO MDI Int: 7
-    SDO AIA 94: 8
-    SDO AIA 131: 9
-    SDO AIA 171: 10
-    SDO AIA 193: 11
-    SDO AIA 211: 12
-    SDO AIA 304: 13
-    SDO AIA 335: 14
-    SDO AIA 1600: 15
-    SDO AIA 1700: 16
-    SDO AIA 4500: 17
-    SDO HMI Int: 18
-    SDO HMI Mag: 19
-    STEREO_A EUVI-A 171: 20
-    STEREO_A EUVI-A 195: 21
-    STEREO_A EUVI-A 284: 22
-    STEREO_A EUVI-A 304: 23
-    STEREO_A COR1-A: 28
-    STEREO_A COR2-A: 29
-    STEREO_B EUVI-B 171: 24
-    STEREO_B EUVI-B 195: 25
-    STEREO_B EUVI-B 284: 26
-    STEREO_B EUVI-B 304: 27
-    STEREO_B COR1-B: 30
-    STEREO_B COR2-B: 31
-    PROBA2 SWAP 174: 32
-    Yohkoh SXT AlMgMn: 33
-    Yohkoh SXT thin-Al: 34
-    Yohkoh SXT white-light: 35
+    >>> # Print a list of datasources and their associated ids
+    >>> ids = {}
+    >>> for name, observatory in datasources.items():  # doctest: +REMOTE_DATA
+    ...          if name == "TRACE":  # doctest: +REMOTE_DATA
+    ...              for wavelength, params in observatory.items():  # doctest: +REMOTE_DATA
+    ...                  ids[params['nickname']] = params['sourceId']  # doctest: +REMOTE_DATA
+    ...          else:  # doctest: +REMOTE_DATA
+    ...              for inst, detectors in observatory.items():  # doctest: +REMOTE_DATA
+    ...                      for wavelength, params in detectors.items():  # doctest: +REMOTE_DATA
+    ...                          if name in ["Hinode", "STEREO_A", "STEREO_B"] or wavelength in ["C2", "C3"]:  # doctest: +REMOTE_DATA
+    ...                              for wave, adict in params.items():  # doctest: +REMOTE_DATA
+    ...                                  ids[name + ' ' + adict['nickname']] = adict['sourceId']  # doctest: +REMOTE_DATA
+    ...                          else:  # doctest: +REMOTE_DATA
+    ...                             ids[name + ' ' + params['nickname']] = params['sourceId']  # doctest: +REMOTE_DATA
+    >>> sorted_ids = sorted(ids.items(), key=lambda x: x[1])  # doctest: +REMOTE_DATA
+    >>> sorted_ids  # doctest: +REMOTE_DATA
```

Comment:
I can't say I'm happy with this how I wrote this example. But it works for now.

Reply:
That looks pretty nasty

Reply:
Now that is good I guess as it has passed all the tests

Reply:
There are only fours tests for this code that don't really test too much. It needs to be overhauled in the future.

Reply:
Oh. The only problem left now is the JPIP stuff. Were we not supporting it from the very beginning?

Reply:
Looks like it breaks if we use it in the code. So we need to fix that or remove it.

Reply:
Um, I don't see the use of using JPIP. That function was written to download a JP2 image and we can download it without using JPIP or any optional parameter

Reply:
Then we should just delete the keyword from the documentation and ignore it for now. When I get other reviews, we can discuss if that was the best choice.

Reply:
Sure I will remove it from the code file.

Reply:
Is it ok now?

Reply:
Yeah.